src/Hub.php

```diff
@@ -44,17 +44,12 @@ class Hub
      */
     public function __construct(array $providers = [], Container $container = null)
     {
-        if ($container === null) {
-            $container = new Container();
-        }
-        $this->container = $container;
+        $this->container = $container === null ? new Container() : $container;
 
         $this->loadLogger();
         $this->loadProviders($providers);
-        $this->logger = $this->container->get(LoggerInterface::class);
```

Comment:
Keep this row. Without this a provider is not able to override the default logger instance used by the Hub class.